Added delta time

Soskar1 opened the pull request:
Added static Time class
Window class updates the static delta_time field of Time class. To accomplish this I added end_frame() method to Window class. Not the best solution, because now it is necessary to call this method at the end of the game loop if you want to update delta_time variable (see FlappyBird or 3D cube examples). I think, it could be fixed with Application layer. For now, we can stay with this solution

Maksasj commented on src/bebone/gfx/window/window.cpp:
```diff
@@ -111,4 +113,10 @@ namespace bebone::gfx {
         return static_cast<f32>(width) / static_cast<f32>(height);
     }
+
+    void Window::end_frame() {
```
![image](https://github.com/Maksasj/bebone/assets/36482401/64009ff9-8a51-402e-bd75-2060bed31371)
maybe this code, and these fields to abstract somehow
```
double start_frame_time;
double end_frame_time;
```

I thought about Timer structure or something like that, idk. What do you think ?

Soskar1 replied:
Ok, I will try to come up with something

Maksasj commented on examples/gfx/opengl/3_opengl_3d_cube/main.cpp:
```diff
@@ -94,12 +94,12 @@ int main() {
         GLContext::clear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
 
-        transform_ptr->rotation = trait_bryan_angle_yxz(Vec3f(t * 0.01f, t * 0.01f, 0.0f));
+        transform_ptr->rotation = trait_bryan_angle_yxz(Vec3f(t, t, 0.0f));
```
Hey, maybe for convenience, lets add Time::running_time(), akka application running time.

![image](https://github.com/Maksasj/bebone/assets/36482401/9beda3de-f838-4327-8863-d8e98ee8df0e)

as I see we have application startup time, so what will not be diffucult

Soskar1 replied:
Yeah, that would be great

Soskar1 replied:
@Maksasj I created [Time::get_seconds_elapsed](https://github.com/Maksasj/bebone/blob/7a746e9304ecb445e811743ed160df386a8e3253/src/bebone/core/timestamp/timestamp.cpp#L19-L22). Maybe naming is not the best, but this method retrevies application running time

Maksasj replied:
> @Maksasj I created [Time::get_seconds_elapsed](https://github.com/Maksasj/bebone/blob/7a746e9304ecb445e811743ed160df386a8e3253/src/bebone/core/timestamp/timestamp.cpp#L19-L22). Maybe naming is not the best, but this method retrevies application running time

nice !

Maksasj reviewed (changes requested):
Nice, but maybe lets add some util things

Soskar1 commented:
Changes:
- Added Timestamp struct. Stores application elapsed time at creation
- Added Watch class. Uses Timestamp struct to calculate time difference

Maksasj reviewed (approved):
nice
